examples/ct/structures_test.go

```diff
@@ -147,6 +149,104 @@ func TestSignV1SCTForPrecertificate(t *testing.T) {
 	assert.Equal(t, cert.RawTBSCertificate, leaf.TimestampedEntry.PrecertEntry.TBSCertificate, "tbs cert mismatch")
 }
 
+func TestSerializeMerkleTreeLeafBadVersion(t *testing.T) {
+	var leaf ct.MerkleTreeLeaf
+
+	leaf.Version = 199 // Out of any expected valid range
+
+	var w io.Writer
+	err := WriteMerkleTreeLeaf(w, leaf)
+	assert.Error(t, err, "incorrectly serialized leaf with bad version")
+}
+
+func TestSerializeMerkleTreeLeafBadType(t *testing.T) {
+	var leaf ct.MerkleTreeLeaf
+
+	leaf.Version = ct.V1
+	leaf.LeafType = 212
+
+	var w io.Writer
+	err := WriteMerkleTreeLeaf(w, leaf)
+	assert.Error(t, err, "incorrectly serialized leaf with bad leaf type")
+}
+
+func TestSerializeMerkleTreeLeafCert(t *testing.T) {
+	ts := ct.TimestampedEntry{Timestamp: 12345, EntryType: ct.X509LogEntryType, X509Entry: ct.ASN1Cert([]byte{0x10, 0x11, 0x12, 0x13, 0x20, 0x21, 0x22, 0x23}), Extensions: ct.CTExtensions{}}
+	leaf := ct.MerkleTreeLeaf{LeafType: ct.TimestampedEntryLeafType, Version: ct.V1, TimestampedEntry: ts}
+
+	var buff bytes.Buffer
```

Comment:
cool.   I assume this grows dynamically as data is written into it.
This is what you'd need to use if you did my comments about all-or-nothing writing above.